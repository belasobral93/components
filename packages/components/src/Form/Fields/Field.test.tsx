/*

 MIT License

 Copyright (c) 2019 Looker Data Sciences, Inc.

 Permission is hereby granted, free of charge, to any person obtaining a copy
 of this software and associated documentation files (the "Software"), to deal
 in the Software without restriction, including without limitation the rights
 to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 copies of the Software, and to permit persons to whom the Software is
 furnished to do so, subject to the following conditions:

 The above copyright notice and this permission notice shall be included in all
 copies or substantial portions of the Software.

 THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 SOFTWARE.

 */

import React from 'react'

import { renderWithTheme } from '@looker/components-test-utils'

import { Field, FieldProps } from './Field'

function getComponentJSX(propsOverride: FieldProps = {}) {
  const defaultProps = { id: 'thumbs-up', label: 'üëç', name: 'thumbsUp' }
  const props = {
    ...defaultProps,
    ...propsOverride,
  }
  return (
    <Field {...props}>
      <input id={props.id} />
    </Field>
  )
}

test('A required Field', () => {
  const { getByText } = renderWithTheme(getComponentJSX({ required: true }))
  expect(getByText('*')).toBeVisible()
})

test('Field supports labelWeight', () => {
  const { container } = renderWithTheme(
    getComponentJSX({ labelFontWeight: 'normal' })
  )
  expect(container).toMatchSnapshot()
})

test('A Field with an error validation aligned to the bottom', () => {
  const { container, getByLabelText, getByText } = renderWithTheme(
    getComponentJSX({
      alignValidationMessage: 'bottom',
      validationMessage: { message: 'This is an error', type: 'error' },
    })
  )
  expect(getByLabelText('üëç')).toBeVisible()
  expect(getByText('This is an error')).toBeVisible()
  expect(container).toMatchSnapshot()
})

test('A Field with an error validation aligned to the left', () => {
  const { container } = renderWithTheme(
    getComponentJSX({
      alignValidationMessage: 'left',
      validationMessage: { message: 'This is an error', type: 'error' },
    })
  )
  expect(container).toMatchSnapshot()
})

test('A Field with an error validation aligned to the right', () => {
  const { container } = renderWithTheme(
    getComponentJSX({
      alignValidationMessage: 'right',
      validationMessage: { message: 'This is an error', type: 'error' },
    })
  )
  expect(container).toMatchSnapshot()
})
