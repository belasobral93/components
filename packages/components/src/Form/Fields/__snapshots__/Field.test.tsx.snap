// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A Field with an error validation aligned to the bottom 1`] = `
.c0 {
  margin-bottom: 0.5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c1 {
  color: #343C42;
  margin-right: 0.5rem;
  font-weight: 600;
  font-size: 0.75rem;
  display: inline-block;
}

.c2 {
  color: #CC1F36;
  margin-right: 0.5rem;
  margin-top: 0.5rem;
  font-size: 0.75rem;
}

<div>
  <div
    class="c0 "
    name="thumbsUp"
  >
    <label
      class="c1"
      color="palette.charcoal700"
      font-size="xsmall"
      font-weight="semiBold"
      for="thumbs-up"
    >
      üëç
    </label>
    <div
      class="c0"
    >
      <div
        class=""
      >
        <input
          id="thumbs-up"
        />
      </div>
      <div
        class="c2"
        font-size="xsmall"
        type="error"
      >
        This is an error
      </div>
    </div>
  </div>
</div>
`;

exports[`A Field with an error validation aligned to the left 1`] = `
.c0 {
  margin-bottom: 0.5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c2 {
  margin-bottom: 0.5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row-reverse;
  -ms-flex-direction: row-reverse;
  flex-direction: row-reverse;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
}

.c1 {
  color: #343C42;
  margin-right: 0.5rem;
  font-weight: 600;
  font-size: 0.75rem;
  display: inline-block;
}

.c3 {
  color: #CC1F36;
  margin-right: 0.5rem;
  margin-top: 0.5rem;
  font-size: 0.75rem;
}

<div>
  <div
    class="c0 "
    name="thumbsUp"
  >
    <label
      class="c1"
      color="palette.charcoal700"
      font-size="xsmall"
      font-weight="semiBold"
      for="thumbs-up"
    >
      üëç
    </label>
    <div
      class="c2"
    >
      <div
        class=""
      >
        <input
          id="thumbs-up"
        />
      </div>
      <div
        class="c3"
        font-size="xsmall"
        type="error"
      >
        This is an error
      </div>
    </div>
  </div>
</div>
`;

exports[`A Field with an error validation aligned to the right 1`] = `
.c0 {
  margin-bottom: 0.5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c2 {
  margin-bottom: 0.5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
}

.c1 {
  color: #343C42;
  margin-right: 0.5rem;
  font-weight: 600;
  font-size: 0.75rem;
  display: inline-block;
}

.c3 {
  color: #CC1F36;
  margin-right: 0.5rem;
  margin-top: 0.5rem;
  margin-left: 0.5rem;
  font-size: 0.75rem;
}

<div>
  <div
    class="c0 "
    name="thumbsUp"
  >
    <label
      class="c1"
      color="palette.charcoal700"
      font-size="xsmall"
      font-weight="semiBold"
      for="thumbs-up"
    >
      üëç
    </label>
    <div
      class="c2"
    >
      <div
        class=""
      >
        <input
          id="thumbs-up"
        />
      </div>
      <div
        class="c3"
        font-size="xsmall"
        type="error"
      >
        This is an error
      </div>
    </div>
  </div>
</div>
`;

exports[`Field supports labelWeight 1`] = `
.c0 {
  margin-bottom: 0.5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c1 {
  color: #343C42;
  margin-right: 0.5rem;
  font-weight: 400;
  font-size: 0.75rem;
  display: inline-block;
}

<div>
  <div
    class="c0 "
    name="thumbsUp"
  >
    <label
      class="c1"
      color="palette.charcoal700"
      font-size="xsmall"
      font-weight="normal"
      for="thumbs-up"
    >
      üëç
    </label>
    <div
      class="c0"
    >
      <div
        class=""
      >
        <input
          id="thumbs-up"
        />
      </div>
    </div>
  </div>
</div>
`;
